### ISSUE TITLE
Replace base58 with a better library

### ISSUE THREAD

**Issue by gz-c** [skycoin/skycoin#762]
See #760 for the original discussion/problem

Our base58 library had a problem with base58 parsing, it treated any leading invalid base58 character as a `0` character (only `"1"` is a valid leading `0` character).

In case it has other problems, it should be replaced with a more readable and maintained/tested library.

This may be a suitable replacement library:
https://github.com/itchyny/base58-go

Before replacing the base58 library, add many valid addresses to the test cases. Compare the two libraries: verify that for valid addresses, the decoded result is the same, and that they encode to the same string values.

**Comment by therealssj** [comment]
Have you taken a look at [this](https://github.com/mr-tron/base58) library
Its fast base58 implementation is significantly faster than of the library mentioned above.

**Comment by gz-c** [comment]
Yes but the code is more complicated so its harder to audit, I'm not sure we should use it

**Comment by therealssj** [comment]
Fair point. 
https://github.com/itchyny/base58-go also provides a neat cli just to quickly test some base58 addresses.

Out of context but you guys have some contributing guidelines?

**Comment by gz-c** [comment]
Regarding contribution guidelines, travis will enforce most of the rules.  Descriptive commit messages or pull requests are helpful

**Comment by therealssj** [comment]
I played with both libraries a bit more and I take back my opinion of going with https://github.com/itchyny/base58-go
The way that library implements the base58 decode is fine but the end result is not the hex value but the ascii value of hex values which makes it more complicated to work as the current codebase is around hex the values.
